### PR TITLE
Create CLI/daemon scaffolding

### PR DESCRIPTION
**Pull request by kubasobon**
## Checklist

- [x] Update changelog in CHANGELOG.md.

### CODE REVIEW

File: cmd/generate/flag.go [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+type flag struct {
+	App          string
+	Dir          string
+	Installation string
+	Version      string
+}
+
+func (f *flag) Init(cmd *cobra.Command) {
+	cmd.Flags().StringVar(&f.App, flagApp, "", `Name of an application to generate the config for (e.g. "kvm-operator").`)
+	cmd.Flags().StringVar(&f.Dir, flagDir, "", "Directory containing configuration. If empty, contents of giantswarm/config will be used.")
```
Comment (kubasobon) [inline cmd/generate/flag.go:24]: Let me know if this is acceptable.

File: cmd/generate/flag.go [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+	flagDir          = "dir"
+	flagInstallation = "installation"
+	flagVersion      = "version"
+)
+
+type flag struct {
+	App          string
+	Dir          string
+	Installation string
+	Version      string
```
Comment (kopiczko) [inline cmd/generate/flag.go:19]: I'd name it ConfigVersion and the flag accordingly because it's confusing.

File: cmd/generate/flag.go (hunk 1 above)
Comment (kopiczko) [reply cmd/generate/flag.go:24]: How about passing a branch instaed?

File: cmd/generate/flag.go (hunk 2 above)
Comment (kubasobon) [reply cmd/generate/flag.go]: Changed.

File: cmd/generate/flag.go (hunk 1 above)
Comment (kubasobon) [reply cmd/generate/flag.go:24]: I thought we can just point it to a dir, regardless of it being git cloned or not. We can update flags when we implement pulling from git anyways.

Comment (kopiczko) [reply cmd/generate/flag.go:24]: That would require yet another implementaiton. When pulling from git we we would be able to find issues with the original implementaiton (at a cost of extra pushing step). If it isn't convincing then let's proceed with --dir.

**Review by kopiczko — APPROVED** [review]

File: cmd/generate/flag.go (hunk 1 above)
Comment (kubasobon) [reply cmd/generate/flag.go]: Changed to `Branch`

**Pull request merged**